Fixed potential exception thrown in multi-episode pattern matching

Warsen opened the pull request:
I discovered this when I was examining the code and debugging tests. If you examine the code, you can see where the Substring function might fail.

Fixed an issue where exception is thrown if multi-episode ending number is matched at the end of a path string.
Removed a conditional operator which is always true.
Removed a fix to get a multi-episode directory path to work as it's no longer necessary.
Changed a line using LINQ functions to use LINQ syntax. It helps with debugging.

Warsen committed:
Fixed an issue where exception is thrown if multi-episode ending number is matched at the end of a path string.
Removed a conditional operator which is always true.
Removed a fix to get a multi-episode directory path to work as it's no longer necessary.
Changed a line using LINQ functions to use LINQ syntax. It helps with debugging.

LukePulverenti commented:
Thanks! Do the unit tests pass? All but 3 should pass, which were already failing.

Warsen commented:
You're welcome, I had fun with it. They do indeed pass except for the 3.

Warsen committed:
Checks for valid season & episode numbers.
Seasons between 200-1927 & greater than 2500 are invalidated.
Episodes above 365 after the 0/1 season are invalidated.
Re-added a fix that adds extension to folder matches because some regular expressions may need it.

Warsen commented:
I added checks for valid episode and season numbers and a few tests which I forgot to mention in the commit, one of them which fails because of it's conflict with a regex pattern.

Episode 105, for example, really should mean episode number 105: not season 1 episode 5. It was always a bad idea idea to have a triple digit number represent a season+episode because it's ambiguous against shows that have had more than 99 episodes in a single year. I don't expect it to be fixed anytime soon but I think it should stay as a test.

LukePulverenti commented on MediaBrowser.Naming.Tests/TV/EpisodeNumberWithoutSeasonTests.cs:
```diff
@@ -17,62 +17,80 @@ public void TestEpisodeNumberWithoutSeason1()
+        [TestMethod]
+        public void TestEpisodeNumberWithoutSeason14()
+        {
+            Assert.AreEqual(317, GetEpisodeNumberFromFile(@"Case Closed (1996-2007)\Case Closed - 317.mkv"));
```
Was this case intended to fail? Because it fails with these changes. Or is the test just wrong? Should it instead be testing for episode 17 instead of 317?

Warsen replied:
Yes, it is indeed supposed to fail. It might sound ambiguous as to whether this is season 3 episode 17 or simply episode 317. It's my opinion that parsing "317" as season 3 episode 17 is a really bad idea, people usually put something to separate the digits like "3x17"
